javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -783,7 +824,51 @@ public boolean isTraversableAgain() {
 
     @Override
     public Iterator<T> iterator() {
-        return isEmpty() ? Iterator.empty() : trie.iterator();
+        return isEmpty() ? Iterator.empty()
+                         : new VectorIterator();
+    }
+
```

Comment:
> Our test cases are not right (interpreted).

I'm surprised you think so. Feel free to investigate, I would be happy to find a better solution :)